pool.py

```diff
@@ -29,20 +26,20 @@ def __init__(self, in_ws_q, out_ws_q, out_display_q):
         self.temp_ambient = TempSensor("28-031724b16bff")  # Ambient
         self.ambient_value = -85
         self.estimate = 0
-        self.heating_state = self.OFF  # ON/OFF/UPKEEP/FOFF
+        self.heating_state = defs.OFF  # ON/OFF/UPKEEP/FOFF
+        self.next_state = defs.OFF
         self.relay = Relay()
         self.magneticValve = MagneticValve()
         self.floatSwitch = FloatSwitch()
         self.pressureSender = PressureSender()
         self.water_level = float(-1)
 
-    def open_valve(self):
-        self.magneticValve.open()
-        return
-
-    def close_valve(self):
-        self.magneticValve.close()
-        return
+    def handle_valve(self):

```

Comment:
This could be structured as a guard clause to reduce arrowness:

```
def handle_valve(self):
  if self.get_water_level() === -1:
    return

  if self.get_water_level() < self.get_water_level_target():
    self.magneticValve.open()
  elif self.get_water_level() >= self.get_water_level_target():
    self.magneticValve.close()
```

Reply:
Sounds good.